api/src/hyp3_api/handlers.py

```diff
@@ -34,7 +35,7 @@ def post_jobs(body, user):
     except QuotaError as e:
         return problem(400, 'Bad Request', str(e))
 
-    request_time = int(time())
+    request_time = datetime.utcnow().isoformat('T') + 'Z'
```

Comment:
https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat

`sep='T'` is the default, don't know that we need to call it out.  I also think we should set `timespec='seconds'`; milisecond precision just seems like a distraction.

`isoformat` gets calls three places in handlers and twice in tests, think it deserves a helper function:

```python
def format_datetime(value: datetime):
    utc_datetime = convert_to_utc(value)  # or implement inline here
    formatted_datetime = utc_datetime.isoformat(timespec='seconds') + 'Z'
    return formatted_datetime
```